### PR TITLE
Change everything to use 'BigDecimal' and add that the 'balance' web socket also responds with a current leader board

### PR DESCRIPTION
**Pull request by loehnertz**
The `balance` WebSocket now sends three messages with this PR.
- balance:{balance}
- rate:{increaseRate}
- leaderboard:{JSON of <String (username) to Number (pokedollars)>}

If this gets merged, the `frontend` has to adapt this change of course.

### CODE REVIEW

**Review by SanderMeister — APPROVED** [review]

File: src/main/kotlin/model/Pokemon.kt [hunk 1]
```diff
@@ -6,20 +8,21 @@ import org.jetbrains.exposed.sql.Table
...
 import service.store.data.ThinPokemon
 import utility.PokeApi
+import java.math.BigDecimal
 
 object Pokemons : Table() {
     val id = integer("id").primaryKey().autoIncrement()
     val pokeNumber = integer("pokeNumber")
     val owner = reference("owner", refColumn = Users.id, onDelete = ReferenceOption.CASCADE)
-    val xp = long("xp")
+    val xp = decimal("xp", 55, 0)
```
Comment (SimonBaars) [inline src/main/kotlin/model/Pokemon.kt:17]: What is the `55`? Maybe make a constant for it.

File: src/main/kotlin/model/User.kt [hunk 2]
```diff
@@ -9,14 +9,15 @@ import org.jetbrains.exposed.sql.select
...
+import java.math.BigDecimal
 
 object Users : Table() {
     val id = integer("id").primaryKey().autoIncrement()
     val name = varchar("name", 255).uniqueIndex()
     val email = varchar("email", 511).uniqueIndex()
     val password = varchar("password", 255)
     val avatarUri = varchar("avatarUri", 511).nullable()
-    val pokeDollars = long("pokeDollars").default(0)
+    val pokeDollars = decimal("pokeDollars", 55, 0).default(BigDecimal(0))
```
Comment (SimonBaars) [inline src/main/kotlin/model/User.kt:20]: Same here. It is probably the size of the decimal.

File: src/main/kotlin/service/store/StoreService.kt [hunk 3]
```diff
@@ -13,16 +13,18 @@ import service.user.balance.BalanceIncreaseRateManager.Companion.IncreaseRateSca
...
 import utility.RedisConnector
+import java.math.BigDecimal
+import java.math.RoundingMode.CEILING
 import kotlin.random.Random
 
 const val BoosterpackSize = 5
 const val LocationIdBaseIncrease = 3.0
 const val SecondsInFiveMinutes = 300
-const val BoosterpackAmountLimit = 25
+const val BoosterpackAmountLimit = 55
```
Comment (SimonBaars) [inline src/main/kotlin/service/store/StoreService.kt:23]: Ah, here it is. Maybe use it everywhere.

**Review by SimonBaars — APPROVED** [review]

File: src/main/kotlin/service/store/StoreService.kt (hunk 3 above)
Comment (loehnertz) [reply src/main/kotlin/service/store/StoreService.kt:23]: It actually does not correspond, I just use 55 oftentimes for pseudo-random numbers :D

**Pull request merged**